lib/dal/src/component.rs

```diff
@@ -3581,10 +3594,45 @@ impl Component {
                 .await?;
             pasted_component_ids.push(pasted_component.id());
             to_pasted_id.insert(component_id, pasted_component.id());
-
+        }
+        let maybe_pasted = |id: ComponentId| to_pasted_id.get(&id).copied().unwrap_or(id);
+        // Copy correct connections (prop to prop and management)
+        for (&component_id, &pasted_component_id) in &to_pasted_id {
             // Copy manager connections
             for manager_id in Component::managers_by_id(ctx, component_id).await? {
-                Component::manage_component(ctx, manager_id, pasted_component.id).await?;
+                // If we were managed by a component that was also pasted, we should be managed by
+                // the pasted version--otherwise we're unmanaged!
+                if pasted_component_ids.contains(&maybe_pasted(manager_id)) {
```

Comment:
You could use `if let Some(&pasted_manager_id) = pasted_component_ids.get(&manager_id) {` here, and then use it on the next line--that would make the intent clear with a few less steps in the way :)

Reply:
And then you probably won't need `maybe_pasted` at all, either :)